src/main/java/org/killbill/billing/plugin/adyen/api/ExpiredPaymentPolicy.java

```diff
@@ -89,4 +91,8 @@ private boolean is3ds(final AdyenPaymentTransactionInfoPlugin transaction) {
         final AdyenResponsesRecord adyenResponsesRecord = transaction.getAdyenResponseRecord().get();
         return PaymentServiceProviderResult.REDIRECT_SHOPPER.toString().equals(adyenResponsesRecord.getResultCode());
     }
+
+    private String getPaymentMethod(final PaymentTransactionInfoPlugin transaction) {
+        return PluginProperties.findPluginPropertyValue("paymentMethod", transaction.getProperties());
```

Comment:
@andrenpaes I think you are correct. I just did a quick check in our `adyen_responses` table and i can see entries without that field and a bit later (after notification) it is populated. So probably we need to handle the case where it is missing (not there yet) and not expire such payment. 

In practice, when notifications are not delayed and based on the fact that janitor will call after minutes (if not hours), we should be good.